### PR TITLE
Fix & tests for bug #10935

### PR DESCRIPTION
**Pull request by SamMousa**
Test case and fix for #10935.

### CODE REVIEW

**Comment by SamMousa** [comment]
Do you guys merge just tests or is nothing ever merged if it contains / causes failing tests?

**Comment by SilverFire** [comment]
No, we do not merge failing tests

**Comment by SamMousa** [comment]
Okay, fixed it.

File: framework/web/UrlManager.php [hunk 1]
```diff
@@ -315,7 +315,12 @@ public function createUrl($params)
         $baseUrl = $this->showScriptName || !$this->enablePrettyUrl ? $this->getScriptUrl() : $this->getBaseUrl();
 
         if ($this->enablePrettyUrl) {
-            $cacheKey = $route . '?' . implode('&', array_keys($params));
+            $cacheKey = $route . '?';
+            foreach($params as $key => $value) {
+                if ($value !== null) {
+                    $cacheKey .= $key . '&';
```
Comment (SilverFire) [inline framework/web/UrlManager.php]: This line causes redundant `&` at the end of the line, but this algorithm is the fastest and it does not matter.

**Comment by SilverFire** [comment]
Merged with minor adjustments. https://github.com/yiisoft/yii2/commit/b6cb1f58a9cc8f7731da24a64137d56e63be1405

**Comment by SilverFire** [comment]
Thank you!

**Pull request closed without merging**